lib/install/deps.js

```diff
@@ -613,8 +635,12 @@ var earliestInstallable = exports.earliestInstallable = function (requiredBy, tr
   // if this tree location requested the same module then we KNOW it
```

Comment:
With an inverted scenario from what I discussed in https://github.com/npm/npm/pull/12811/files#r64309610: ```
pkg@1.0.0 
├── DEV: depA@^1.2.0
├─┬ libA@1.0.0
│ ├── depA@~1.2.0
```
A transitive prod dep can change what dev dep version gets installed. This would mean that:
```
npm install --only=dev && npm install --only=prod
```

Would give you:
```
pkg@1.0.0 
├── depA@1.4.5
├─┬ libA@1.0.0
│ ├── depA@1.2.7
```
But:
```
npm install --only=prod && npm install --only=dev
```
would give you:
```
pkg@1.0.0 
├── depA@1.2.7
├── libA@1.0.0
```
But I'm a lot more comfortable with that lack of symmetry than the other kind.